### ISSUE TITLE
Since v3.53 bundle installation via Kar-file dropping into $install-dir/deploy seems no longer to work

### ISSUE THREAD

**Issue by gschafra** [sonatype/nexus-public#106]
# What problem are you trying to solve/occured?
After uprading Nexus to v3.53 in an Docker based deployment (see below, installation via dropping into `$install-dir/deploy` according to https://help.sonatype.com/repomanager3/integrations/bundle-development/installing-bundles#InstallingBundles-InstallingBundlesInto3.15.0andLater) , the composer community plugin does not longer work nor is loaded. See issue https://github.com/sonatype-nexus-community/nexus-repository-composer/issues/140. 

# Do you have a workaround you are using at present?
Unfortunately not 😱... and Nexus does not allow downgrades!

# How could we solve this issue? (Not knowing is okay!)
Please help... we're running into serious issues here....

# Tell us about your Nexus Repository deployment: what version, operating system, and database are you using?
 - Docker v23 on Debian 11 with `sonatype/nexus3:latest` image (v3.53)
 - see https://github.com/sonatype-nexus-community/nexus-repository-composer/blob/master/Dockerfile

**Comment by joedragons** [comment]
To be clear, the issue here is that the composer plugin doesn't work (basically a duplicate of https://github.com/sonatype-nexus-community/nexus-repository-composer/issues/140) and _not_ that your instance doesn't start.  I suspect if the latter issue it could be solved by (temporarily) removing the plugin, so wanted to make sure.

**Comment by gschafra** [comment]
> To be clear, the issue here is that the composer plugin doesn't work (basically a duplicate of [sonatype-nexus-community/nexus-repository-composer#140](https://github.com/sonatype-nexus-community/nexus-repository-composer/issues/140)) and _not_ that your instance doesn't start?

A new installed (bare/blank) instance starts without any problems, but the dropped plugin is not shown to be loaded/activated according to the logs

> I suspect if the latter issue it could be solved by (temporarily) removing the plugin, so wanted to make sure.

Unfortunately, since the composer repo plugin already run successfully on our old nexus v3.52 and a corresponding repository has been set up, the instance (Docker based!) doesn't start any more, showing following errors in the logs:

```
nexus3-nexus-1  | 2023-05-03 15:31:18,600+0000 ERROR [FelixStartLevel] *SYSTEM org.sonatype.nexus.repository.manager.internal.RepositoryManagerImpl - Failed transition: NEW -> STARTED
nexus3-nexus-1  | java.lang.IllegalStateException: Missing recipe: composer-proxy
nexus3-nexus-1  |       at com.google.common.base.Preconditions.checkState(Preconditions.java:591)
nexus3-nexus-1  |       at org.sonatype.nexus.repository.manager.internal.RepositoryManagerImpl.recipe(RepositoryManagerImpl.java:159)
nexus3-nexus-1  |       at org.sonatype.nexus.repository.manager.internal.RepositoryManagerImpl.newRepository(RepositoryManagerImpl.java:177)
nexus3-nexus-1  |       at org.sonatype.nexus.repository.manager.internal.RepositoryManagerImpl.restoreRepositories(RepositoryManagerImpl.java:274)
nexus3-nexus-1  |       at org.sonatype.nexus.repository.manager.internal.RepositoryManagerImpl.doStart(RepositoryManagerImpl.java:256)
nexus3-nexus-1  |       at org.sonatype.nexus.common.stateguard.StateGuardLifecycleSupport.start(StateGuardLifecycleSupport.java:69)
nexus3-nexus-1  |       at org.sonatype.nexus.common.stateguard.MethodInvocationAction.run(MethodInvocationAction.java:39)
nexus3-nexus-1  |       at org.sonatype.nexus.common.stateguard.StateGuard$TransitionImpl.run(StateGuard.java:193)
nexus3-nexus-1  |       at org.sonatype.nexus.common.stateguard.TransitionsInterceptor.invoke(TransitionsInterceptor.java:57)
nexus3-nexus-1  |       at org.sonatype.nexus.extender.NexusLifecycleManager.startComponent(NexusLifecycleManager.java:199)
nexus3-nexus-1  |       at org.sonatype.nexus.extender.NexusLifecycleManager.to(NexusLifecycleManager.java:111)
nexus3-nexus-1  |       at org.sonatype.nexus.extender.NexusContextListener.moveToPhase(NexusContextListener.java:319)
nexus3-nexus-1  |       at org.sonatype.nexus.extender.NexusContextListener.frameworkEvent(NexusContextListener.java:216)
nexus3-nexus-1  |       at org.apache.felix.framework.Felix.setActiveStartLevel(Felix.java:1597)
nexus3-nexus-1  |       at org.apache.felix.framework.FrameworkStartLevelImpl.run(FrameworkStartLevelImpl.java:308)
nexus3-nexus-1  |       at java.lang.Thread.run(Thread.java:750)
nexus3-nexus-1  | 2023-05-03 15:31:18,605+0000 ERROR [FelixStartLevel] *SYSTEM org.sonatype.nexus.extender.NexusContextListener - Failed to start nexus
nexus3-nexus-1  | java.lang.IllegalStateException: Missing recipe: composer-proxy
nexus3-nexus-1  |       at com.google.common.base.Preconditions.checkState(Preconditions.java:591)
nexus3-nexus-1  |       at org.sonatype.nexus.repository.manager.internal.RepositoryManagerImpl.recipe(RepositoryManagerImpl.java:159)
nexus3-nexus-1  |       at org.sonatype.nexus.repository.manager.internal.RepositoryManagerImpl.newRepository(RepositoryManagerImpl.java:177)
nexus3-nexus-1  |       at org.sonatype.nexus.repository.manager.internal.RepositoryManagerImpl.restoreRepositories(RepositoryManagerImpl.java:274)
nexus3-nexus-1  |       at org.sonatype.nexus.repository.manager.internal.RepositoryManagerImpl.doStart(RepositoryManagerImpl.java:256)
nexus3-nexus-1  |       at org.sonatype.nexus.common.stateguard.StateGuardLifecycleSupport.start(StateGuardLifecycleSupport.java:69)
nexus3-nexus-1  |       at org.sonatype.nexus.common.stateguard.MethodInvocationAction.run(MethodInvocationAction.java:39)
nexus3-nexus-1  |       at org.sonatype.nexus.common.stateguard.StateGuard$TransitionImpl.run(StateGuard.java:193)
nexus3-nexus-1  |       at org.sonatype.nexus.common.stateguard.TransitionsInterceptor.invoke(TransitionsInterceptor.java:57)
nexus3-nexus-1  |       at org.sonatype.nexus.extender.NexusLifecycleManager.startComponent(NexusLifecycleManager.java:199)
nexus3-nexus-1  |       at org.sonatype.nexus.extender.NexusLifecycleManager.to(NexusLifecycleManager.java:111)
nexus3-nexus-1  |       at org.sonatype.nexus.extender.NexusContextListener.moveToPhase(NexusContextListener.java:319)
nexus3-nexus-1  |       at org.sonatype.nexus.extender.NexusContextListener.frameworkEvent(NexusContextListener.java:216)
nexus3-nexus-1  |       at org.apache.felix.framework.Felix.setActiveStartLevel(Felix.java:1597)
nexus3-nexus-1  |       at org.apache.felix.framework.FrameworkStartLevelImpl.run(FrameworkStartLevelImpl.java:308)
nexus3-nexus-1  |       at java.lang.Thread.run(Thread.java:750)
nexus3-nexus-1  | 2023-05-03 15:31:18,608+0000 ERROR [FelixStartLevel] *SYSTEM Felix - Framework listener delivery error.
nexus3-nexus-1  | java.lang.IllegalStateException: Missing recipe: composer-proxy
nexus3-nexus-1  |       at com.google.common.base.Preconditions.checkState(Preconditions.java:591)
nexus3-nexus-1  |       at org.sonatype.nexus.repository.manager.internal.RepositoryManagerImpl.recipe(RepositoryManagerImpl.java:159)
nexus3-nexus-1  |       at org.sonatype.nexus.repository.manager.internal.RepositoryManagerImpl.newRepository(RepositoryManagerImpl.java:177)
nexus3-nexus-1  |       at org.sonatype.nexus.repository.manager.internal.RepositoryManagerImpl.restoreRepositories(RepositoryManagerImpl.java:274)
nexus3-nexus-1  |       at org.sonatype.nexus.repository.manager.internal.RepositoryManagerImpl.doStart(RepositoryManagerImpl.java:256)
nexus3-nexus-1  |       at org.sonatype.nexus.common.stateguard.StateGuardLifecycleSupport.start(StateGuardLifecycleSupport.java:69)
nexus3-nexus-1  |       at org.sonatype.nexus.common.stateguard.MethodInvocationAction.run(MethodInvocationAction.java:39)
nexus3-nexus-1  |       at org.sonatype.nexus.common.stateguard.StateGuard$TransitionImpl.run(StateGuard.java:193)
nexus3-nexus-1  |       at org.sonatype.nexus.common.stateguard.TransitionsInterceptor.invoke(TransitionsInterceptor.java:57)
nexus3-nexus-1  |       at org.sonatype.nexus.extender.NexusLifecycleManager.startComponent(NexusLifecycleManager.java:199)
nexus3-nexus-1  |       at org.sonatype.nexus.extender.NexusLifecycleManager.to(NexusLifecycleManager.java:111)
nexus3-nexus-1  |       at org.sonatype.nexus.extender.NexusContextListener.moveToPhase(NexusContextListener.java:319)
nexus3-nexus-1  |       at org.sonatype.nexus.extender.NexusContextListener.frameworkEvent(NexusContextListener.java:216)
nexus3-nexus-1  |       at org.apache.felix.framework.Felix.setActiveStartLevel(Felix.java:1597)
nexus3-nexus-1  |       at org.apache.felix.framework.FrameworkStartLevelImpl.run(FrameworkStartLevelImpl.java:308)
nexus3-nexus-1  |       at java.lang.Thread.run(Thread.java:750)
```

**Comment by gschafra** [comment]
Might be related to `[KARAF-7306] - Hot deployment (deploy directory) does not work for provisioned blueprints`
as solved in https://github.com/apache/karaf/blob/main/RELEASE-NOTES.md#apache-karaf-440?

**Comment by joedragons** [comment]
I suspect this is because you have composer repositories still but I am not sure how to remove them without a running instance.  I suspect the composer community is better suited to help you quickly or https://community.sonatype.com than this site, but I'll let others tell me I'm wrong.

**Comment by gschafra** [comment]
> I suspect the composer community is better suited to help you quickly or https://community.sonatype.com/ than this site, but I'll let others tell me I'm wrong.

Sorry, I don't think so. Please take my reference to the Karaf bug into account (see above). The goal must be that the plugin, which was provided according to your instructions, worked in the last nexus version and has not changed even since that version, works again. I am convinced you can do it 😉. I would really appreciate it, if you help me with this ❤

**Comment by gschafra** [comment]
As suspected... the problem also occurs with another third-party bundle/plugin: https://github.com/sonatype-nexus-community/nexus-repository-cargo. So, it's DEFINITLY NOT related the the bundle/plugin itself. This strengthens me in the assumption that it is a problem of Nexus itself.

**Comment by bguerin** [comment]
I am having this issue too, with both https://github.com/sonatype-nexus-community/nexus-repository-cargo and https://github.com/asharapov/nexus-casc-plugin

**Comment by nblair** [comment]
Thanks for the report @gschafra - we're taking a look at this now. While we do not provide direct support for those specific plugins, I am able to replicate the behavior you've described with another plugin.

**Comment by nblair** [comment]
We've just published an advisory on the downloads page regarding 3.53.0: https://help.sonatype.com/repomanager3/product-information/download

We have an internal ticket opened and have escalated appropriately.

**Comment by nblair** [comment]
Update: we've isolated the root cause to a subtle change in how Apache Karaf is configured in 3.53. We are reviewing a change now.

**Comment by rtista** [comment]
I've been suffering from this problem as well. Is there any fix release scheduled?

**Comment by nblair** [comment]
We are testing a fix now, it's likely to be included in our 3.54.0 release scheduled for availability on Tuesday May 23.

**Comment by nblair** [comment]
We just confirmed the fixed on this issue, so expect it to see the fix in 3.54 next week. Thanks for bringing this to our attention!
